common/flatpak-run.c

```diff
@@ -2963,8 +2963,49 @@ flatpak_run_app (const char     *app_ref,
 
   if (app_deploy != NULL)
     {
+      g_autofree const char **previous_ids = NULL;
+      gsize len = 0;
+
       app_files = flatpak_deploy_get_files (app_deploy);
 
+      g_autoptr (GFile) current_data_dir = flatpak_get_data_dir (app_ref_parts[1]);
+      previous_ids = flatpak_deploy_data_get_previous_ids (app_deploy_data, &len);
+      if (previous_ids)
```

Comment:
Because len starts as 0, we can actually skip this if - the loop will be skipped if len is 0. This will reduced the indentation.